tests/transformers/test_transformers_model_export.py

```diff
@@ -1389,7 +1389,49 @@ def test_text2text_generation_pipeline_with_inference_configs(
     assert dict_inference == inference
 
 
-def test_text2text_generation_pipeline_with_params(text2text_generation_pipeline, tmp_path):
+def test_text2text_generation_pipeline_with_inference_config_and_params(
+    text2text_generation_pipeline, tmp_path
+):
+    data = "muppet keyboard type"
+    inference_config = {
+        "top_k": 2,
+        "num_beams": 5,
+        "top_p": 0.85,
+        "repetition_penalty": 1.15,
+    }
+    parameters = {"top_k": 3, "max_length": 30, "temperature": 0.62}
+    generated_output = mlflow.transformers.generate_signature_output(
+        text2text_generation_pipeline, data
+    )
+    signature = infer_signature(
+        data,
+        generated_output,
+        parameters,
+    )
+
+    model_path = tmp_path / "model"
+    mlflow.transformers.save_model(
+        text2text_generation_pipeline,
+        path=model_path,
+        inference_config=inference_config,
+        signature=signature,
+    )
+    pyfunc_loaded = mlflow.pyfunc.load_model(model_path)
+
+    # inference_config and default params are all applied
+    res = pyfunc_loaded.predict(data)
+    applied_params = inference_config.copy()
+    applied_params.update(parameters)
+    res2 = pyfunc_loaded.predict(data, applied_params)
+    assert res == res2
```

Comment:
But res and res2 are predicted using the same parameters including temperature, would that be flaky as well?

Reply:
Temperature, if set above the default of `0`, introduces a degree of non-determinism and a non-linear 
degree of probability of variation in the output of the selected retrieved next token when iterating through the generative algorithm. If the temperature were `0` here and other params were changed, I wouldn't see any issue with the approach here. It's not an issue with the input data here or the current state of the model, but if this model gets updated again and the test data becomes more responsive to dynamic generation probabilities, it will be a confusing and frustrating bug to track down and fix.

Reply:
Thanks for the detailed explanation Ben! I'll remove temperature override in params. Because I want to make sure all parameters used for predict are the same, so a single string check seems not enough, and looks like removing temperature can make the response steady :)